### ISSUE TITLE
Avoid zero_grad()

### ISSUE THREAD

**Issue by cowwoc** [PyTorchLightning/pytorch-lightning#9503]
https://ai.plainenglish.io/best-performance-tuning-practices-for-pytorch-3ef06329d5fe section `Zeroing The Gradients Efficiently` proposes an alternative that they claim is more efficient than invoking `zero_grad()`. Has this been tried before?

**Comment by ananthsub** [comment]
There's a `set_to_none` flag as an option on the optimizer's zero_grad: https://pytorch.org/docs/stable/generated/torch.optim.Optimizer.zero_grad.html https://pytorch.org/docs/stable/_modules/torch/optim/optimizer.html#Optimizer.zero_grad

Have you tried using this?
